spec/services/medication_reminders_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "spec_helper"
+require "rails_helper"
+
+describe MedicationReminders do
+  let(:deliveries) { ActionMailer::Base.deliveries }
+
+  before { ActionMailer::Base.deliveries = [] }
+
+  describe "#send_take_medication_reminder_emails" do
+    context "no reminders" do
+      it "sends no emails" do
+        MedicationReminders.new.send_take_medication_reminder_emails
+        expect(deliveries.count).to eq(0)
+      end
+    end
+
+    context "reminders exist" do
+      let!(:user) { FactoryGirl.create(:user1) }
+      let!(:medication) { FactoryGirl.create(:medication, userid: user.id) }
+      let!(:reminder) { FactoryGirl.create(:take_medication_reminder, medication_id: medication.id) }
+
+      context "take medication reminder is active" do
+        it "sends an email" do
+          MedicationReminders.new.send_take_medication_reminder_emails
+          expect(deliveries.count).to eq(1)
+        end
+      end
+
+      context "take medication reminder is not active" do
+        it "doesn't send an email" do
+          reminder.update(active: false)
+          MedicationReminders.new.send_take_medication_reminder_emails
+          expect(deliveries.count).to eq(0)
+        end
+      end
+    end
+  end
+
+  describe "#send_refill_reminder_emails" do
+    context "no reminders" do
+      it "sends no emails" do
+        MedicationReminders.new.send_refill_reminder_emails
+        expect(deliveries.count).to eq(0)
+      end
+    end
+
+    context "reminders exist and it is one week before the refill" do
+      let!(:user) { FactoryGirl.create(:user1) }
```

Comment:
Could this be a build_stubbed instead of a create? It looks like it's only used for its id, so it doesn't need to be persisted (I think).

Another option would be to add
`association :user, factory: :user1` to the medication factory, but we might not always need the associated user.